### ISSUE TITLE
'unique constraint violated' when using email as preferred username claim with Google OIDC

### ISSUE THREAD

**Issue by Milkshak3s** [SecurityBrewery/catalyst#707]
My setup that generated this error utilizes Google's OIDC provider for authentication into Catalyst.
(https://accounts.google.com/.well-known/openid-configuration)
(https://support.google.com/cloud/answer/6158849?hl=en)

### Problem
When attempting to log in with `accounts.google.com` configured as the OIDC provider, and `email` as the OIDC username, catalyst responds first with `JSONErrorStatus 500 could not load user` and then `could not create user: HTTPError(409): unique constraint violated` on a second attempt.

### To reproduce
1. Set up a catalyst to authenticate via Google OIDC
2. Set `OIDC_CLAIM_USERNAME` to `email`
3. Start catalyst (`docker compose up` or your configuration)
4. Navigate to the catalyst page. Click "Log in with OIDC"
5. Experience a redirect to the login page.
6. See the following error on the backend:
```
api.go:28: JSONErrorStatus 500 could not load user: HTTPError(404): document not found
```
7. Click "login" once again.
8. See the following error message:
```
{"error":"could not create user: HTTPError(409): unique constraint violated - in index primary of type primary over '_key'; conflicting key: my-email@domain-com"}
```

**Comment by Milkshak3s** [comment]
This is a bit of an issue as there aren't too many options beyond `sub` to use as a username, as `name` provides spaces that generate an `illegal document key` error.

**Comment by jonas-plum** [comment]
Should be solved in [v0.10.2](https://github.com/SecurityBrewery/catalyst/releases/tag/v0.10.2)
